hera_qm/omnical_metrics.py

```diff
@@ -12,3 +28,796 @@ def get_omnical_metrics_dict():
+
+        tot_phs_std : float
+            average of ant_phs_std
+
+        phs_std_band_ants : ndarray
+            array containing antennas whose phs_std didn't meet phs_std_cut
+
+        phs_std_good_sol : bool
+            boolean with metric's good / bad determination of entire solution
+        """
+        # take robust standard deviation
+        ant_phs_std = np.sqrt(np.array(map(astats.biweight_midvariance, phs_diff[:, :, self.band, :])))
+        tot_phs_std = np.median(ant_phs_std)
+
+        # get goodness of variability
+        phs_std_bad_ants = self.ant_array[np.where(ant_phs_std > phs_std_cut)]
+        phs_std_good_sol = tot_phs_std < phs_std_cut
+
+        if return_dict == True:
+            ant_phs_std = OrderedDict(zip(self.ant_array, ant_phs_std))
+
+        return ant_phs_std, tot_phs_std, phs_std_bad_ants, phs_std_good_sol
+
+    def plot_chisq_metric(self, ax=None, save=False, fname=None, **kwargs):
+        """
+        Plot Chi-Square output from omnical for each antenna
+
+        Input:
+        ------
+        ax : matplotlib axis object, default=None
+
+        save : bool, default=False
+            if True, save image as a png
+
+        fname : str, default=None
+            filename to save image as
+
+        outpath : str, default=None
+            path to place file in
+            will default to location of *omni.calfits file
+        """
+        if hasattr(self, 'metrics') == False:
+            raise Exception("Must run self.run_metrics() before plotting routines...")
+
+        plot_chisq_metric(self.metrics, ax=ax, save=save, fname=fname, **kwargs)
+
+    def plot_phs_metric(self, plot_type='std', ax=None, save=False, fname=None, outpath=None, **kwargs):
+        """
+        Plot omnical phase metric
+
+        Input:
+        ------
+        plot_type : str, default='std', options=['std', 'ft']
+            specify plot type
+
+            'std' plots stand deviations of omni-firstcal phase difference.
+            large stand deviations are not ideal
+
+            'ft' plots the "noise_level"
+            see self.phs_FT_metric for details.
+
+        ax : matplotlib axis object
+
+        save : bool, default=False
+            if True, save image as a png
+
+        fname : str, default=None
+            filename to save image as
+
+        outpath : str, default=None
+            path to place file in
+            will default to location of *omni.calfits file
+        """
+        if hasattr(self, 'metrics') == False:
+            raise Exception("Must run self.run_metrics() before plotting routines...")
+
+        if self.firstcal_file is None:
+            raise Exception("Must supply firstcal_file in order to plot phase metrics...")
+
+        plot_phs_metric(self.metrics, plot_type=plot_type, ax=ax, save=save, fname=fname, outpath=outpath, **kwargs)
+
+    def plot_gains(self, ants=None, time_index=0, jones_index=0, divide_fc=False, plot_type='phs', ax=None,
```

Comment:
Why are some of the plotting routines functions outside the class, with simple wrappers in the class, while others (like this one) only live in the class?

Reply:
the plotting routines that live outside the class can be used with a generic `metrics` dictionary output. the plotting routines that live inside the class require data that doesn't come with a generic `metrics` dictionary, and therefore require one to make an instantiation of `OmniCal_Metics`. 

For example, the `plot_gains` function requires the `self.gain_diff` array, which is a very large array, so I don't put it into the `metrics` dictionary.